### PR TITLE
[GraphQL] handle subscription efficiently and move meshsync subscription to headers

### PR DESCRIPTION
**Pull request by bariqhibat**
**Description**

This PR fixes #

**Notes for Reviewers**
Introduces:
- Remove MeshSync subscription in Dashboard component and move it to Headers
- Dispose subscription in Dashboard efficiently (earlier, it closes all the subscription client)
- Need https://github.com/meshery/meshery/pull/4689 to establish broker subscription in header as well @leecalcote

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

https://user-images.githubusercontent.com/54849895/144631478-5e68ed5c-e65c-4211-ba44-8ee188d91e21.mp4

### CODE REVIEW

**Comment by bariqhibat** [comment]
It seems like the current UI E2E test includes the subscription for the meshsync to be in the Dashboard component. Here, we shift that to the headers so that it is included in every single route, can you take a look? @sudo-NithishKarthik @sayantan1413 @warunicorn19

**Comment by warunicorn19** [comment]
Sure, taking a look @bariqhibat

**Comment by warunicorn19** [comment]
@bariqhibat rn it's 2 red circles right?

**Comment by sudo-NithishKarthik** [comment]
> It seems like the current UI E2E test includes the subscription for the meshsync to be in the Dashboard component. Here, we shift that to the headers so that it is included in every single route, can you take a look? @sudo-NithishKarthik @sayantan1413 @warunicorn19

who is working on this ?
@Prakhar-Agarwal-byte  @Abhishek-kumar09  @warunicorn19

**Comment by warunicorn19** [comment]
Probably something we should deal with right away, i wont be available till Monday. If it isn't done by then, then I can take a look.

**Comment by Prakhar-Agarwal-byte** [comment]
@sudo-NithishKarthik I will be working on a UI part here after it is merged. Reference [#1](https://layer5io.slack.com/archives/C023QGMSH8D/p1639070694122100), [#2](https://layer5io.slack.com/archives/C023QGMSH8D/p1639123433125400)

**Comment by bariqhibat** [comment]
Commented out the UI part!

**Comment by bariqhibat** [comment]
Should we merge this? @sudo-NithishKarthik

**Review by warunicorn19 — APPROVED** [review]
Yup, I think this is good to merge now.

**Review by sayantan1413 — APPROVED** [review]
Yes this can be merged ✔️

File: ui/components/Header.js [hunk 1]
```diff
@@ -136,6 +175,15 @@ class Header extends React.Component {
                   <div data-test="notification-button">
                     <MesheryNotification />
                   </div>
+
+                  {/* <Tooltip title="MeshSync Status">
```
Comment (leecalcote) [inline ui/components/Header.js:179]: @Prakhar-Agarwal-byte just a reminder that this is where you'll insert your MeshSync logo logic.

**Review by leecalcote — APPROVED** [review]

File: ui/components/Header.js [hunk 2]
```diff
@@ -63,6 +67,41 @@ const styles = (theme) => ({
 });
 
 class Header extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      meshSyncStatusSubscription : null,
```
Comment (leecalcote) [inline ui/components/Header.js:73]: @sudo-NithishKarthik, @Abhishek-kumar09  - - just a brief note that the state of these objects will need to be accounted for during the UI refactoring - with redux.

**Review by leecalcote — APPROVED** [review]

File: ui/components/Header.js (hunk 1 above)
Comment (Prakhar-Agarwal-byte) [reply ui/components/Header.js:179]: @leecalcote Added MeshSync logo in #4785

**Pull request merged**